llvm/lib/Object/RelocationResolver.cpp

```diff
@@ -310,6 +310,28 @@ static uint64_t resolveX86(uint64_t Type, uint64_t Offset, uint64_t S,
   }
 }
 
+static bool supportsXtensa(uint64_t Type) {
+  switch (Type) {
+  case ELF::R_XTENSA_NONE:
+  case ELF::R_XTENSA_32:
+    return true;
+  default:
+    return false;
+  }
+}
+
+static uint64_t resolveXtensa(uint64_t Type, uint64_t Offset, uint64_t S,
+                              uint64_t LocData, int64_t Addend) {
+  switch (Type) {
+  case ELF::R_XTENSA_NONE:
```

Comment:
We could drop `R_*_NONE`. It should not appear in relocation files.

Reply:
i feel it's better for this kind of tools to handle more cases as far as it's legal and easy.
anyway your comment isn't specific to this PR, right?

Reply:
No, the comment is specific to this PR.

RelocationResolver is to handle static relocations. While R_*_NONE might be defined by a psABI, in practice it's always a toolchain bug when RelocationResolver has to resolve R_*_NONE.  We should not handle unnecessary relocations.

Reply:
> Yes, the comment is specific to this PR.
> 
> RelocationResolver is to handle static relocations. While R___NONE might be defined by a psABI, in practice it's always a toolchain bug when RelocationResolver has to resolve R___NONE. We should not handle unnecessary relocations.

do you mean that for some reasons your rationale is specific to xtensa (this PR)
and doesn't apply to other non-xtensa relocations like R_X86_64_NONE?

Reply:
@MaskRay ping

even if it's an indication of toolchain bugs, i feel it's nicer for llvm-dwarfdump to be able to dump it.

Reply:
No. DWARF sections should not contain NONE relocations.

Reply:
can you please explain why it's specific to this PR and doesn't apply to R_X86_64_NONE etc?

Reply:
Not specific to this PR. We could remove R_X86_64_NONE. Here R_X86_64_NONE is not the best example to follow. You should follow other architectures without NONE.